fpdcc/ccfp-asset-dashboard
# Improve funding models

## Overview

Includes changes to the models related to phase and funding. Adds a form for the new `FundingStream` model. Adds more information about funding to the phase detail page.

For issues:
- #71 
- #74 
- #75 

## Demo
<img width="1407" alt="Screen Shot 2022-03-09 at 9 41 25 AM" src="https://user-images.githubusercontent.com/38969506/157476476-987d65b6-2a3c-4084-99af-41ee67c793d4.png">
<img width="1405" alt="Screen Shot 2022-03-09 at 9 41 34 AM" src="https://user-images.githubusercontent.com/38969506/157476887-d6be6115-9b3b-4bcd-ba0f-fa8130460cbd.png">


### Notes
The UI for the funding form is especially awkward because it doesn't have many fields and spans across the entire page (see second image above). We could maybe make that a modal form on the phase form page, do you have any thoughts @fgregg?

## Testing Instructions
- visit review app: https://ccfp-asset-d-funding-mo-4urajt.herokuapp.com/
- sign in with the `ccfp-asset-dashboard test user` credentials in last pass
- go to a phase. 
- add funding. make sure it shows up as expected after creating the funding instance.
- edit the funding and make sure it reloads the correct data.
- add another funding stream and make sure it renders in the table.
- go back to the phase list view for the project and make sure a total of all your funding shows up in the table.
